File: voom-fe/src/app/shared/rest/vehicle/vehicle-type.api.ts
import {Api} from '../api';
import {ApiClient} from '../api-client';
import {RequestConfig} from '../rest.model';
import {CreateVehicleTypeDto, VehicleTypeDto} from './vehicle-type.model';

export default class VehicleTypeApi extends Api {

  constructor(apiClient: ApiClient) {
    super(apiClient);
  }

  getVehicleTypes() {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.get<void, VehicleTypeDto[]>('/api/vehicleTypes', config);
  }

  getVehicleType(id: number) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.get<void, VehicleTypeDto>(`/api/vehicleTypes/${id}`, config);
  }

  createVehicleType(vehicleTypeDto: CreateVehicleTypeDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.post<CreateVehicleTypeDto, VehicleTypeDto>('/api/vehicleTypes', vehicleTypeDto, config);
  }

  updateVehicleType(id: number, vehicleTypeDto: VehicleTypeDto) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json',
        contentType: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.put<VehicleTypeDto, VehicleTypeDto>(`/api/vehicleTypes/${id}`, vehicleTypeDto, config);
  }

  deleteVehicleType(id: number) {
    const config: RequestConfig = {
      headers: {
        accept: 'application/json'
      },
      authenticated: true,
    };

    return this.apiClient.delete<void, void>(`/api/vehicleTypes/${id}`, config);
  }
}

File: voom-app/app/src/main/java/inc/visor/voom/app/shared/api/VehicleTypeApi.java
package inc.visor.voom.app.shared.api;

import java.util.List;

import inc.visor.voom.app.shared.dto.vehicle.CreateVehicleTypeDto;
import inc.visor.voom.app.shared.dto.vehicle.VehicleTypeDto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface VehicleTypeApi {

    @GET("api/vehicleTypes")
    Call<List<VehicleTypeDto>> getVehicleTypes();

    @GET("api/vehicleTypes/{id}")
    Call<VehicleTypeDto> getVehicleType(@Path("id") long id);

    @POST("api/vehicleTypes")
    Call<VehicleTypeDto> createVehicleType(@Body CreateVehicleTypeDto dto);

    @PUT("api/vehicleTypes/{id}")
    Call<VehicleTypeDto> updateVehicleType(@Path("id") long id, @Body VehicleTypeDto dto);

    @DELETE("api/vehicleTypes/{id}")
    Call<Void> deleteVehicleType(@Path("id") long id);

}
